### PR TITLE
Rewrite KS test in python

### PR DESCRIPTION
**Pull request by kurtwheeler**
## Issue Number

https://github.com/AlexsLemonade/refinebio/issues/2878

## Purpose/Implementation Notes

This uses python to run a KS test rather than passing the whole matrix to R. We still use rpy2 to quantile normalize but we're already doing it in chunks so it shouldn't be too bad of a RAM-suck.

I've also reenabled the qn_test for compendia and set them back to public when they're created.

## Methods

@jaclyn-taroni it's probably appropriate for you to take a look to make sure I didn't miss any subtleties of the R code, but I think it was pretty straightforward.

## Types of changes

- New feature (non-breaking change which adds functionality)

## Functional tests

Compendia and smasher tests are still passing! Or they were, they may have stopped, still investigating

## Checklist

_Put an `x` in the boxes that apply._

- [ ] Lint and unit tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)
- [ ] Any dependent changes have been merged and published in downstream modules

## Screenshots

Please attach any screenshots that illustrate these changes.

### CODE REVIEW

File: workers/data_refinery_workers/processors/smashing_utils.py [hunk 1]
```diff
@@ -509,52 +511,33 @@ def _test_qn(merged_matrix):
...
+    if num_columns < 2:
         return None
 
-    # This wont work with larger matricies
-    # https://github.com/AlexsLemonade/refinebio/issues/1860
-    ncolumns = ncol(merged_R_matrix)
-
-    if ncolumns[0] <= 200:
+    if num_columns <= 200:
         # Convert to NP, Shuffle, Return to R
```
Comment (jaclyn-taroni) [inline workers/data_refinery_workers/processors/smashing_utils.py:522]: This comment needs updating

File: workers/data_refinery_workers/processors/smashing_utils.py [hunk 2]
```diff
@@ -509,52 +511,33 @@ def _test_qn(merged_matrix):
...
+    random.seed(123)
+    num_columns = merged_matrix.shape[1]
     # Not enough columns to perform KS test - either bad smash or single sample smash.
-    if n < m:
+    if num_columns < 2:
         return None
 
-    # This wont work with larger matricies
-    # https://github.com/AlexsLemonade/refinebio/issues/1860
-    ncolumns = ncol(merged_R_matrix)
```
Comment (jaclyn-taroni) [inline workers/data_refinery_workers/processors/smashing_utils.py:533]: Do we still have this problem, such that we need the logic below, now that we're not passing things to R? We probably don't know _yet_ do we?

**Review by jaclyn-taroni — COMMENTED** [review]
Doesn't look to me like you missed any subtleties or R code!

**Review by davidsmejia — APPROVED** [review]
This looks much cleaner. nice job!

File: workers/data_refinery_workers/processors/smashing_utils.py (hunk 2 above)
Comment (kurtwheeler) [reply workers/data_refinery_workers/processors/smashing_utils.py:533]: I don't think we do, but I would be surprised if we did.

**Pull request merged**